### PR TITLE
refactor HeightChange, make it easy for adding new heights in future

### PR DESCRIPTION
**Pull request by dustinxie**
HeightChange is defined by heights in config file, so move its definition and code from EVM params to config.

To add a new height in future, we only need to add in config yaml, and modify a single file, no need to change EVM params function interface

### CODE REVIEW

File: config/heightchange.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+// IsPost return true if height is after the height change
+func (hc *HeightChange) IsPost(height uint64, name HeightName) bool {
+	var h uint64
+	if name == Pacific {
+		h = hc.pacificHeight
+	} else if name == Aleutian {
+		h = hc.aleutianHeight
+	} else {
+		panic("invalid height name!")
```
Comment (zjshen14) [inline config/heightchange.go]: log.Panic

File: config/heightchange.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	}
+	return height >= h
+}
+
+// IsPre return true if height is before the height change
+func (hc *HeightChange) IsPre(height uint64, name HeightName) bool {
```
Comment (zjshen14) [inline config/heightchange.go]: This method is not necessary

File: config/heightchange.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	Pacific = iota
+	Aleutian
+)
+
+type (
+	// HeightName is codename for height changes
+	HeightName int
+
+	// HeightChange lists heights at which certain fixes take effect
+	HeightChange struct {
```
Comment (zjshen14) [inline config/heightchange.go]: rename it to UpgradeHeight?

File: action/protocol/execution/evm/evmstatedbadapter.go [hunk 4]
```diff
@@ -55,15 +56,15 @@ type (
 		preimageSnapshot map[int]preimageMap
 		dao              db.KVStore
 		cb               db.CachedBatch
-		hc               *HeightChange
+		hc               *config.HeightChange
```
Comment (zjshen14) [inline action/protocol/execution/evm/evmstatedbadapter.go]: Let's avoid using ptr

File: config/heightchange.go (hunk 2 above)
Comment (dustinxie) [reply config/heightchange.go]: in the first code change, i don't have it, but then i found
!hc.IsPost(), you might miss the ! at front
so add this IsPre() for better readability

**Comment by dustinxie** [comment]
> Good refactor.
> 
> Please search for PacificHeight and refactor the relevant code as well

yes, that's done

**Pull request merged**